src/pages/Debug/DebugDetails.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+import type {OnyxKey} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import Button from '@components/Button';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+
+type DebugDetailsProps = {
+    data: Record<string, unknown>;
+    onyxKey: OnyxKey;
+    isCollection?: boolean;
+    idSelector?: (data: Record<string, unknown>) => string;
+};
+
+function DebugDetails({data, onyxKey, isCollection = false, idSelector = () => ''}: DebugDetailsProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [draftData, setDraftData] = useState(data);
+    const [errors, setErrors] = useState<Record<string, string>>({});
+    const [hasChanges, setHasChanges] = useState<boolean>(false);
+    return (
+        <ScrollView
+            style={styles.mt5}
+            contentContainerStyle={[styles.gap5, styles.ph5, styles.pb5]}
+        >
+            {Object.entries(data ?? {}).map(([key, value]) => (
+                <TextInput
+                    errorText={errors[key]}
+                    accessibilityLabel="Text input field"
+                    key={key}
+                    forceActiveLabel
+                    label={key}
+                    numberOfLines={4}
+                    multiline={typeof value === 'object'}
+                    defaultValue={typeof value === 'object' ? JSON.stringify(value, null, 6) : (value as string)}
+                    onChangeText={(updatedValue) => {
+                        setDraftData((currentDraftData) => {
+                            return {...currentDraftData, [key]: updatedValue};
+                        });
+                        setHasChanges(true);
+                    }}
+                />
+            ))}
+            <Text style={[styles.headerText, styles.textAlignCenter]}>{translate('debug.hint')}</Text>
+            <Button
+                success
+                text={translate('common.save')}
+                isDisabled={!hasChanges}
+                onPress={() => {
```

Comment:
This is where I think we should decouple -- to create a new `onSave` prop where each component that uses `DebugDetails` will have its own implementation, that way I think we can simplifly some logic here.